hazelcast/src/main/java/com/hazelcast/quorum/impl/QuorumImpl.java

```diff
@@ -188,14 +189,20 @@ boolean isPingAware() {
     }
 
     /**
-     * Returns if quorum is needed for this operation. This is determined by the {@link QuorumConfig#type} and by the type
-     * of the operation - {@link ReadonlyOperation} or {@link MutatingOperation}.
+     * Returns if quorum is needed for this operation.
+     * The quorum is determined by the {@link QuorumConfig#type} and by the type of the operation -
+     * {@link ReadonlyOperation} or {@link MutatingOperation}.
      *
      * @param op the operation which is to be executed
      * @return if this quorum should be consulted for this operation
      * @throws IllegalArgumentException if the quorum configuration type is not handled
      */
     private boolean isQuorumNeeded(Operation op) {
+        if (op instanceof QuorumCheckAwareOperation
```

Comment:
I think we should check this only if otherwise - based on the type of the operation and the quorum - the quorum is needed. The current implementations are free, but if future implementations of the `shouldCheckQuorum()` methods are doing any logic, we may waste some cycles unnecessarily, regardless of the operation and the quorum type, e.g. `MutatingOperation` and `READ` quorum.

Reply:
Addressed review comment, please verify.